packages/ra-ui-materialui/src/button/DeleteWithConfirmButton.tsx

```diff
@@ -147,14 +126,66 @@ const DeleteWithConfirmButton = ({
     );
 };
 
+const sanitizeRestProps = ({
+    handleSubmit,
+    handleSubmitWithRedirect,
+    invalid,
+    label,
+    pristine,
+    saving,
+    submitOnEnter,
+    undoable,
+    ...rest
+}: DeleteWithConfirmButtonProps) => rest;
+
+const useStyles = makeStyles(
+    theme => ({
+        deleteButton: {
+            color: theme.palette.error.main,
+            '&:hover': {
+                backgroundColor: fade(theme.palette.error.main, 0.12),
+                // Reset on mouse devices
+                '@media (hover: none)': {
+                    backgroundColor: 'transparent',
+                },
+            },
+        },
+    }),
+    { name: 'RaDeleteWithConfirmButton' }
+);
+
+interface Props {
```

Comment:
Lots of duplication with DeleteButton. Couldn't we improve that ?

Reply:
That'd imply extracting the common part to another file (to avoid circular dependencies), and that would make all the Delete components harder to read... Again, I prefer some duplication for clarity.